web/yo/app/scripts/directives/curationQueue.js

```diff
@@ -0,0 +1,142 @@
+                    });
+                    scope.dtOptions = {
+                        hasBootstrap: true,
+                        paging: false,
+                        scrollCollapse: true,
+                        aaSorting: [[0, 'asc']]
+                    };
+                    scope.dtColumns = [
+                        DTColumnDefBuilder.newColumnDef(0),
+                        DTColumnDefBuilder.newColumnDef(1),
+                        DTColumnDefBuilder.newColumnDef(2),
+                        DTColumnDefBuilder.newColumnDef(3).withOption('sType', 'date'),
+                        DTColumnDefBuilder.newColumnDef(4),
+                        DTColumnDefBuilder.newColumnDef(5).withOption('sType', 'curated'),
+                        DTColumnDefBuilder.newColumnDef(6)
+                    ];
+                },
+                post: function postLink(scope) {
+                    scope.$watch('article', function(n, o) {
+                        if(n && n !== o) {
+                            $timeout.cancel(scope.articleTimeoutPromise);
+                            scope.articleTimeoutPromise = $timeout(function() {
+                                if(/^[\d]*$/.test(scope.article)) {
+                                    scope.getArticle(scope.article);
+                                }
+                            }, 500);
+                        }
+                    });
+                }
+            },
+            controller: function($scope) {
+                $scope.addCuration = function() {
+                    var item = $rootScope.model.createMap();
+                    item.set('link', $scope.link);
+                    item.set('variant', $scope.variant);
+                    item.set('curator', $scope.curator);
+                    item.set('curated', false);
+                    item.set('addedBy', user.name);
+                    item.set('addedAt', new Date().getTime());
+                    if($scope.predictedArticle && $scope.validPMID) {
+                        item.set('article', $scope.predictedArticle);
+                        item.set('pmid', $scope.article);
+                    } else {
+                        item.set('article', $scope.article);
+                    }
+                    $scope.queueModel.push(item);
+                    $scope.queue.push({article: item.get('article'), pmid: item.get('pmid'), link: item.get('link'), variant: item.get('variant'), addedBy: item.get('addedBy'), addedAt: item.get('addedAt'), curated: item.get('curated'), curator: item.get('curator')});
+                    $scope.article = '';
+                    $scope.link = '';
+                    $scope.variant = '';
+                    $scope.curator = '';
+                    $scope.predictedArticle = '';
+                    $scope.validPMID = false;
+                };
+                $scope.editCuration = function(index) {
+                    $scope.queue[index].editable = true;
```

Comment:
Ideally, we should validate index. But I don't see any chance this will be undefined if this function is only called in html. But in case we may use this function in the future within js file. Let's still validate the index